app/assets/javascripts/govuk_publishing_components/components/autocomplete.js

```diff
@@ -0,0 +1,461 @@
+//= require ../vendor/polyfills/closest.js
+/* global fetch */
+
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {}
+window.GOVUK.Modules.GovukAutocomplete = window.GOVUKFrontend.Autocomplete;
+
+(function (Modules) {
+  'use strict'
+
+  function GemAutocomplete ($module) {
```

Comment:
I notice that across the code in here we've got some parts embracing es6 (asnyc, const, arrow functions) and other bits that are es5 (function.prototype classes, var) it'd be good to just consistently write it in es6 code.